Adding mention of a known bug and its workaround (#9897)
flutter/website: src/platform-integration/ios/app-extensions.md

description: Learn how to add app extensions to your Flutter apps
---

iOS app extensions allow you to expand functionality
outside your app. Your app could appear as a home screen widget,
or you can make portions of your app available within other apps.

To learn more about app extensions, check out
[Apple's documentation][].

{{site.alert.note}}
  If you experience a build error when building an
  iOS app that includes an app extension, be
  aware that there is an open bug. The workaround
  involves changing the order of the build process.
  For more information, check out

>   [Issue #9690][] and [Issue #135056][].

Reviewer: **question:** Can we provide more information on the workaround? If so, can we include it on the page? It seems not user friendly to throw them to issues to figure it out.

Author: I disagree, Tony. The website bug concisely and clearly explains the workaround.

Reviewer 2: Redirecting developers to the issue tracker has the added advantage that the developers will always have the latest information. Bugs have the tendency to get fixed, at which point the workaround will no longer be required.

    </div>
</figure>

As of the 3.16 release, you can build
Flutter UI for an app extension,
though you must use an extension-safe
`Flutter.xcframework` and embed the
`FlutterViewController` as described in
the following section.

{{site.alert.note}}
  Due to the memory limitations of app extensions,
  it's only recommended to use Flutter to build app extension UI
  for extension types that have memory limits larger than 100MB.
  For example, share extensions has a 120MB memory limit.

  In addition, Flutter uses extra memory in debug mode.

>   Therefore, Flutter doesn't fully support running
>   app extensions in debug mode on physical devices
>   when used to build extension UI. As an alternative,
>   use an iOS simulator to test your extension in debug mode.

Reviewer: **issue:** This needs to be stated more clearly. Does Flutter run app extensions in debug mode or not? If so, write the difficulty that might be experienced when doing so. If it doesn't, say it doesn't run it.

Consider something more direct:

```suggestion
  Therefore, when used to build the iOS extension UI in debug mode,
  Flutter doesn't run iOS app extensions on physical devices.
  As an alternative, use an iOS simulator to test your extension in debug mode.
```

Author: It depends on the size of the app.

{{site.alert.end}}

1. Locate the extension-safe `Flutter.xcframework` file,
   at `<path_to_flutter_sdk>/bin/cache/artifacts/engine/ios/extension_safe/Flutter.xcframework`.
   

>     * To build for release or profile mode, find the

Reviewer: **nitpick:** 

```suggestion
    * To build for release or profile modes, find the
```

>       framework file under the `ios-profile` or `ios-release`

Reviewer: **nitpick:** Switch the two folders because you used "respectively".
```suggestion
      framework file under the `ios-release` or `ios-profile`
```

       </div>
   </figure>

1. Open the Flutter app project settings in Xcode
   to share build configurations. 

> 

Reviewer: **suggestion:** Add the lowercase alphabetical ordering.

```suggestion

    {: type="a"}
```

1. Build and run the main application target.
1. After the app is launched on the simulator,
   press <kbd>Cmd</kbd> + <kbd>Shift</kbd> + <kbd>H</kbd>
   to minimize the app, which switches to the home screen.

> 1. Launch an app that supports the Share Extension,

Reviewer: **question:** Should Share Extension be capitalized throughout? If not, is it UI text? If that's the case, bold it?

Author: No, it shouldn't be Share Extension, but I missed a couple. It should be either "Share extension" or "share extension" depending on context.
